c2corg_ui/static/js/imageuploader.js

```diff
@@ -227,7 +233,8 @@ app.ImageUploaderController.prototype.uploadFile_ = function(file) {
     'metadata': {
       'id': file['name'] + '-' + new Date().toISOString(),
       'activities': [],
-      'categories': []
+      'categories': [],
+      'image_type': this.image_type_
```

Comment:
because it will execute the function every time you upload an image whereas `this.image_type_` is calculated once at the beginning (i hope) ?